src/Draco.Compiler.Tests/Syntax/ParserTests.cs

```diff
@@ -114,4 +204,1269 @@ public void TestEmptyFuncWithoutClosingCurly()
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Literal>();
+                        {
+                            this.T(TokenType.LiteralInteger);
+                        }
+                    }
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x = 5;", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationNoTypeValueMissingValue()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Unexpected>();
+                        {
+                            this.T(TokenType.Semicolon);
+                        }
+                    }
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x =;", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationNoValueType()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<TypeSpecifier>();
+                    {
+                        this.T(TokenType.Colon);
+                        this.N<TypeExpr.Name>();
+                        {
+                            this.T(TokenType.Identifier);
+                        }
+                    }
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x: int32;", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationNoValueTypeMissingType()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<TypeSpecifier>();
+                    {
+                        this.T(TokenType.Colon);
+                        this.N<TypeExpr.Name>();
+                        {
+                            this.MissingT(TokenType.Identifier);
+                        }
+                    }
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x:;", Declaration);
+    }
+
+    [Fact]
+    public void TestDeclarationValueType()
```

Comment:
Again, name is inexact, also don't be afraid to type out words like With or Without.